lms/templates/student_account/register.underscore

```diff
@@ -29,11 +29,17 @@
             <%
             _.each( context.providers, function( provider) {
                 if ( provider.registerUrl ) { %>
-                    <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
-                        <span class="icon fa <%- provider.iconClass %>" aria-hidden="true"></span>
-                        <span aria-hidden="true"><%- provider.name %></span>
-                        <span class="sr"><%- _.sprintf( gettext("Create account using %(providerName)s."), {providerName: provider.name} ) %></span>
-                    </button>
+                    <div class="button-container" <% if ( provider.buttonColor ) { %>style="color: <%- provider.buttonColor %>; background-color: <%- provider.buttonColor %>"<% } %>>
+                        <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
+                            <div class="icon <% if ( provider.iconClass ) { %>fa <%- provider.iconClass %><% } %>" <% if ( provider.buttonColor ) { %>style="background: <%- provider.buttonColor %>"<% } %> aria-hidden="true">
+                                <% if ( provider.iconImage ) { %>
+                                    <img src="<%- provider.iconImage %>" alt="<%- provider.name %> icon" />
+                                <% } %>
+                            </div>
+                            <span <% if ( provider.textColor ) { %>style="color: <%- provider.textColor %>"<% } %> aria-hidden="true"><%- provider.name %></span>
```

Comment:
Users can add arbitrary/custom providers\*, so they need to have *some* mechanism to specify the color/style and icon of their custom providers. Our current approach is just that they specify the icon and colors to use when configuring the provider, but @AlasdairSwan is suggesting we find a way to use sass to control the appearance of any custom buttons. As far as I know, that would require each instance that wants to use custom providers to style their buttons within a theme, right?

\* e.g. Imagine I want to add GitHub login on my instance, or SAML login to my local university.